dist_ir/executor/simulator.py

```diff
@@ -50,22 +50,27 @@ def __init__(self, function: Function, inputs: Sequence[Any]):
                     isinstance(inputs[i], ConcreteValue)
                     and inputs[i].device is not None
                 ):
-                    self.peak_memory[inputs[i].device] += inputs[i].val.nbytes
+                    cur_live_memory = self.live_memory[inputs[i].device][0][-1]
```

Comment:
Would it be simpler/cleaner if we used `self.update_live_memory` for these updates?